slf4cats-impl/src/main/scala/slf4cats/impl/ContextLogger.scala

```diff
@@ -139,49 +129,45 @@ object ContextLogger {
     override def withArg[A](
         name: String,
         value: => A,
-        toJson: Option[A => String] = None,
-    ): ContextLogger[F] =
+    )(implicit logEncoder: LogEncoder[A]): ContextLogger[F] =
       withComputed(
         name,
         FAsync.delay {
           value
         },
-        toJson,
       )
 
     override def withComputed[A](
         name: String,
         value: F[A],
-        toJson: Option[A => String] = None,
-    ): ContextLogger[F] = {
+    )(implicit logEncoder: LogEncoder[A]): ContextLogger[F] = {
       val memoizedJson =
         Async.memoize(
-          value.flatMap(JsonInString.make(toJson.getOrElse(toJsonGlobal))(_)),
+          value.flatMap(JsonInString.make(logEncoder.encode)(_)),
         )
       new ContextLoggerImpl[F](
         underlying,
         context + ((name, memoizedJson)),
-        toJsonGlobal,
       )
     }
 
-    override def withArgs[A](
-        map: Map[String, A],
-        toJson: Option[A => String] = None,
-    ): ContextLogger[F] = {
-      val toJsonLocal = toJson.getOrElse(toJsonGlobal)
-      new ContextLoggerImpl[F](
-        underlying,
-        context ++ map
-          .mapValues(v =>
-            Async.memoize(
-              JsonInString
-                .make(toJsonLocal)(v),
-            ),
-          ),
-        toJsonGlobal,
-      )
-    }
+//    override def withArgs[A](
+//        map: Map[String, A],
+//        toJson: Option[A => String] = None,
+//    ): ContextLogger[F] = {
+//      val toJsonLocal = toJson.getOrElse(toJsonGlobal)
+//      new ContextLoggerImpl[F](
+//        underlying,
+//        context ++ map
+//          .mapValues(v =>
+//            Async.memoize(
+//              JsonInString
+//                .make(toJsonLocal)(v),
+//            ),
+//          ),
+//        toJsonGlobal,
+//      )
+//    }
```

Comment:
Is it intentional, that `withArgs` is removed? I think that such a method could be useful from time to time.

Reply:
Thought to emulate it with LogEncoder. But in the end I think separate method is better. Will uncomment.